### PR TITLE
GEOMESA-2859 Fix Avro user data serialization to match schema

### PR DESCRIPTION
**Pull request by elahrvivaz**
Signed-off-by: Emilio Lahr-Vivaz <elahrvivaz@ccri.com>

### CODE REVIEW

File: geomesa-features/geomesa-feature-avro/src/test/scala/org/locationtech/geomesa/features/avro/AvroSimpleFeatureWriterTest.scala [hunk 1]
```diff
@@ -115,19 +115,19 @@ class AvroSimpleFeatureWriterTest extends Specification with Mockito with Abstra
...
       there was one(encoder).writeString(vis)
 
       // hint data
       there was one(encoder).writeString(HintKeySerialization.keyToId(Hints.USE_PROVIDED_FID))
-      there was one(encoder).writeBoolean(true)
+      there was one(encoder).writeString("true")
 
       // key = 5, value = null
-      there was one(encoder).writeInt(5)
+      there was one(encoder).writeString("5")
```
Comment (jnh5y) [inline geomesa-features/geomesa-feature-avro/src/test/scala/org/locationtech/geomesa/features/avro/AvroSimpleFeatureWriterTest.scala:130]: Hmm... why did this change from Int to String?  (Just finishing my first skim through.)

Comment (elahrvivaz) [reply geomesa-features/geomesa-feature-avro/src/test/scala/org/locationtech/geomesa/features/avro/AvroSimpleFeatureWriterTest.scala:130]: the avro user-data schema is defined as only being strings

Comment (jnh5y) [reply geomesa-features/geomesa-feature-avro/src/test/scala/org/locationtech/geomesa/features/avro/AvroSimpleFeatureWriterTest.scala:130]: Ah.... and that may explain the pendingUntilFixed thing in the unit test...

**Comment by jnh5y** [comment]
Merged as 4cce58abc56e3ddd0800a28d2ec0d3ae62c9f8fd

**Pull request closed without merging**